src/SimulatedEffect/Http.elm

```diff
@@ -0,0 +1,322 @@
+                    onResult (Err Http.Timeout)
+
+                Http.NetworkError_ ->
+                    onResult (Err Http.NetworkError)
+
+                Http.BadStatus_ metadata body ->
+                    onResult (Err <| Http.BadStatus metadata.statusCode)
+
+                Http.GoodStatus_ _ body ->
+                    onResult (Ok body)
+
+
+{-| Expect the response body to be JSON.
+-}
+expectJson : (Result Http.Error a -> msg) -> Decoder a -> Expect msg
+expectJson onResult decoder =
+    Expect <|
+        \response ->
+            case response of
+                Http.BadUrl_ s ->
+                    onResult (Err <| Http.BadUrl s)
+
+                Http.Timeout_ ->
+                    onResult (Err Http.Timeout)
+
+                Http.NetworkError_ ->
+                    onResult (Err Http.NetworkError)
+
+                Http.BadStatus_ metadata _ ->
+                    onResult (Err <| Http.BadStatus metadata.statusCode)
+
+                Http.GoodStatus_ _ body ->
+                    case Json.Decode.decodeString decoder body of
+                        Err jsonError ->
+                            onResult (Err <| Http.BadBody <| Json.Decode.errorToString jsonError)
+
+                        Ok value ->
+                            onResult (Ok value)
+
+
+{-| Expect the response body to be whatever.
+-}
+expectWhatever : (Result Error () -> msg) -> Expect msg
+expectWhatever onResult =
+    Expect <|
+        \response ->
+            case response of
+                Http.BadUrl_ s ->
+                    onResult (Err <| Http.BadUrl s)
+
+                Http.Timeout_ ->
+                    onResult (Err Http.Timeout)
+
+                Http.NetworkError_ ->
+                    onResult (Err Http.NetworkError)
+
+                Http.BadStatus_ metadata _ ->
+                    onResult (Err <| Http.BadStatus metadata.statusCode)
+
+                Http.GoodStatus_ _ _ ->
+                    onResult (Ok ())
+
+
+{-| -}
+type alias Error =
+    Http.Error
+
+
+{-| -}
+type alias Response body =
+    Http.Response body
+
+
+{-| Just like [`request`](#request), but it creates a `Task`.
+-}
+task :
+    { method : String
+    , headers : List Header
+    , url : String
+    , body : Body
+    , resolver : Resolver x a
+    , timeout : Maybe Float
+    }
+    -> SimulatedTask x a
+task r =
+    SimulatedEffect.HttpTask
+        { method = r.method
+        , url = r.url
+        , headers = r.headers
+        , body =
+            case r.body of
+                EmptyBody ->
+                    ""
+
+                StringBody body ->
+                    body.content
+        , onRequestComplete =
+            case r.resolver of
+                StringResolver f ->
+                    f
+        }
+
+
+{-| Describes how to resolve an HTTP task.
+-}
+type Resolver x a
+    = StringResolver (Response String -> SimulatedTask x a)
+
+
+{-| Turn a response with a `String` body into a result.
```

Comment:
What's the intended external use of this resolver? The comment feels sparse

Reply:
This is just paralleling the function in elm/http: https://package.elm-lang.org/packages/elm/http/latest/Http#stringResolver  (but produces a SimulatedEffect.Http.Expect instead of an Http.Expect) -- the full docs are in elm/http